betatest/urls.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+URL Patterns for the `betatest` app
+"""
+
+# Imports #####################################################################
+
+from django.conf.urls import url
+
+from betatest import views
+
+
+# URL Patterns ################################################################
+
+app_name = 'beta'
+urlpatterns = [
+    url(r'^register/$', views.BetaTestApplicationView.as_view(), name='register'),
+    url(r'^success/$', views.success, name='success'),
+    url(r'^api/register/validate/$', views.validate_registration, name='validate'),
```

Comment:
Yes this makes it discoverable through the browsable api web interface.

I'm not happy with the coupling between apps, as it means that we can no longer move the app into its own django project, for example. But that is not why this feels like a hack to me.

It feels like a hack because DRF routers only support viewsets, which are an abstraction designed to automatically generate a RESTful api for models. In this case we are simply validating a form, models are not involved. So to get this to work with the browsable api I had to repurpose the viewset's `list` method, even though we are not listing any models. That feels like a hack to me because we are not using the abstraction the way it was designed. It just happens to work.

The quick and easy way I have done this in the past is to simply list the available endpoints in the project's documentation. Then you can visit the url in your browser and see DRF's autogenerated docs.

A more complete solution would be to use something like [DRF Docs](https://github.com/ekonstantinidis/django-rest-framework-docs). Other options are listed here: http://www.django-rest-framework.org/topics/documenting-your-api/